feat: support aten.atan2 converter

chohk88 opened the pull request:
# Description

New feature to support aten.atan2 converter. I also add test case including edge case for inputs are zero. 

Fixes # ([issue](https://github.com/pytorch/TensorRT/issues/2586))

## Type of change

Please delete options that are not relevant and/or add your own.
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

zewenli98 reviewed (commented):
Overall looks good to me. It seems that `get_value_by_condition` is a helper function. Can you move it to `.../dynamo/conversion/converter_utils.py`?

gs-olive commented on py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:
```diff
@@ -213,6 +216,178 @@ def remainder(
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    other: TRTTensor,
```
Does this function support `other` being a Python float or does it require `TRTTensor`?

chohk88 replied:
I think it requires `TRTTensor` because both input and output are `Tensor`

![image](https://github.com/pytorch/TensorRT/assets/25674295/007179b0-024e-44fe-93da-6d754d63bf14)

gs-olive commented on py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:
```diff
@@ -556,3 +731,17 @@ def le(
     )
+
+
+def get_value_by_condition(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    other: TRTTensor,
+    condition: TRTTensor,
+) -> TRTTensor:
+    select_layer = ctx.net.add_select(condition, input, other)
+    set_layer_name(select_layer, target, name + "_select", source_ir)
+    return select_layer.get_output(0)
```
This can be removed - replace with usage of `impl.condition.select`, from here:
https://github.com/pytorch/TensorRT/blob/76c9ebdc7544f7f666b63e21d13abfc9e2d44c5d/py/torch_tensorrt/dynamo/conversion/impl/condition/ops.py#L102-L113

gs-olive replied:
Alternatively, you may need `where`:
https://github.com/pytorch/TensorRT/blob/76c9ebdc7544f7f666b63e21d13abfc9e2d44c5d/py/torch_tensorrt/dynamo/conversion/impl/condition/ops.py#L18-L26

chohk88 replied:
I removed `get_value_by_condition` and replaced it with `impl.condition.select.`

gs-olive commented on py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:
```diff
@@ -1391,6 +1391,24 @@ def aten_ops_atanh(
     )
 
 
+@dynamo_tensorrt_converter(torch.ops.aten.atan2.default)
```
If you require one or both input tensors to be `TRTTensor`, consider using:
https://github.com/pytorch/TensorRT/blob/6b8ccbc69f05de3b6275cac4613735934fb559b0/py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py#L62-L66

chohk88 replied:
I add `enforce_tensor_types` for both input and other.

chohk88 commented:
@zewenli98 @gs-olive I've made the changes you suggested. I also noticed I missed the case when input or other is a single constant, so I fixed that using `broadcast` function. And, I added test cases to check it's all working right. Thanks for your helpful suggestions!

narendasan reviewed (approved):
LGTM
